src/lib/marshal/workflow/types.ts

```diff
@@ -54,6 +55,20 @@ type BatchStepData = WorkflowStepBase & {
   settings: BatchStepSettings;
 };
 
+/* Throttle step */
+
+type ThrottleStepSettings = {
+  throttle_window?: Duration;
+  throttle_window_field_path?: string;
+  throttle_key?: string;
+  throttle_limit?: number;
```

Comment:
```suggestion
  throttle_limit: number;
```
Nit, if we always provide a default and `validate_required` on it, then probably just type it non optional?